src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class BaseDataConditionValidator(CamelSnakeSerializer):
+
+    condition = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Condition used to compare data value to the stored comparison value",
+    )
+
+    @property
+    def comparison(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def condition_result(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def type(self) -> str:
+        # TODO: This should probably at least be an enum
+        raise NotImplementedError
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        attrs["type"] = self.type
+        return attrs
+
+
+class NumericComparisonConditionValidator(BaseDataConditionValidator):
+
+    comparison = serializers.FloatField(
+        required=True,
+        help_text="Comparison value to be compared against value from data.",
+    )
+    result = serializers.ChoiceField(
+        choices=[
+            (DetectorPriorityLevel.HIGH, "High"),
+            (DetectorPriorityLevel.MEDIUM, "Medium"),
+            (DetectorPriorityLevel.LOW, "Low"),
+        ]
+    )
+
+    @property
+    def supported_conditions(self) -> frozenset[Condition]:
+        raise NotImplementedError
+
+    @property
+    def supported_results(self) -> frozenset[DetectorPriorityLevel]:
+        raise NotImplementedError
+
+    def validate_condition(self, value: str) -> Condition:
+        try:
+            condition = Condition(value)
+        except ValueError:
+            condition = None
+
+        if condition not in self.supported_conditions:
+            raise serializers.ValidationError(f"Unsupported condition {value}")
+        return condition
+
+    def validate_result(self, value: str) -> DetectorPriorityLevel:
+        try:
+            result = DetectorPriorityLevel(int(value))
+        except ValueError:
+            result = None
+        if result not in self.supported_results:
+            raise serializers.ValidationError("Unsupported condition result")
+        return result
+
+
+class BaseDataSourceValidator(CamelSnakeSerializer):
+    pass
+
+
+class BaseGroupTypeDetectorValidator(CamelSnakeSerializer):
+    name = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Name of the uptime monitor",
+    )
+    group_type = serializers.CharField()
+
+    def validate_group_type(self, value: str) -> type[GroupType]:
+        detector_type = grouptype.registry.get_by_slug(value)
+        if detector_type is None:
+            raise serializers.ValidationError("Unknown group type")
+        if detector_type.detector_validator is None:
+            raise serializers.ValidationError("Group type not compatible with detectors")
+        # TODO: Probably need to check a feature flag to decide if a given
+        # org/user is allowed to add a detector
```

Comment:
🤔 I feel like ideally we have the feature flag gating in the API itself, thoughts?

Reply:
We need both - in the future, we might be adding new detector types and want to restrict access to them